src/integrations/bare.ts

```diff
@@ -63,14 +63,13 @@ export class ExpoBareIntegration {
       // The name of the sourcemap file in Sentry is different depending on whether it was uploaded
       // by the upload-sourcemaps script in this package (in which case it will have a revisionId)
       // or by the default @sentry/react-native script.
-      let sentryFilename: string;
-
-      sentryFilename = Platform.OS === 'android' ? 'index.android.bundle' : 'main.jsbundle';
-
-      error.stack = error.stack.replace(
-        /\/(bundle\-\d+|[\dabcdef]+\.bundle)/g,
-        `/${sentryFilename}`
-      );
+      if (typeof error.stack === 'string') {
+        const sentryFilename = Platform.OS === 'android' ? 'index.android.bundle' : 'main.jsbundle';
+        error.stack = error.stack.replace(
```

Comment:
```suggestion
        error.stack = error.stack?.replace(
```
With this conditional, you don't need the outer `if` statement, and the call is safe.

Reply:
The current logic checks whether `error.stack` is actually a string, whereas this only checks if it's defined. In most cases it's a relatively safe assumption that if `error.stack` is defined it's also a string, but my goal here was to make sure that we're not making _any_ assumptions here (eg. `throw { stack: [] }` is totally valid).